### PR TITLE
Vesting routes check

### PR DESCRIPTION
**Pull request by theghost1980**

### CODE REVIEW

**Review by stoodkev — CHANGES_REQUESTED** [review]
Small design changes requested on Clickup:
https://app.clickup.com/t/86epqn71h

File: src/components/carousel/carousel.tsx [hunk 1]
```diff
@@ -37,16 +21,36 @@ interface Props {
...
-const Carousel = ({buttonsConfig, content, locale, theme}: Props) => {
+const Carousel = ({
+  buttonsConfig,
+  theme,
+  carouselContent,
+  renderItem,
+  hideButtons,
+  moveNext,
+  resetMoveNext,
+  enableSwap,
```
Comment (stoodkev) [inline src/components/carousel/carousel.tsx:41]: should be `enableSwipe`

File: src/components/carousel/carousel.tsx [hunk 2]
```diff
@@ -37,16 +21,36 @@ interface Props {
...
+  moveNext?: number;
+  resetMoveNext?: () => void;
+  enableSwap?: boolean;
 }
 
-const Carousel = ({buttonsConfig, content, locale, theme}: Props) => {
+const Carousel = ({
+  buttonsConfig,
+  theme,
+  carouselContent,
```
Comment (stoodkev) [inline src/components/carousel/carousel.tsx:36]: we're in the carousel, so you can just call this `content`

File: src/components/carousel/carousel.tsx [hunk 3]
```diff
@@ -37,16 +21,36 @@ interface Props {
     lastTitle: string;
     lastSlideAction?: () => void | any;
   };
-  content: WhatsNewContent;
-  locale: string;
+  carouselContent: any[];
+  renderItem: (item: any) => React.JSX.Element;
   theme: Theme;
+  hideButtons?: boolean;
+  moveNext?: number;
```
Comment (stoodkev) [inline src/components/carousel/carousel.tsx:28]: From what I understand of your code, this should be a boolean, I only see 0 and 1 as potential values. Though passing a flag to signal you should move to the next index is quite convoluted. A better way would be to create a reference of the carousel on the parent, and allow the parent to call sth like `carouselRef.moveToNext();`

**Review by stoodkev — CHANGES_REQUESTED** [review]

File: src/components/carousel/carousel.tsx (hunk 1 above)
Comment (theghost1980) [reply src/components/carousel/carousel.tsx:41]: moved as point to fix to clickup

File: src/components/carousel/carousel.tsx (hunk 3 above)
Comment (theghost1980) [reply src/components/carousel/carousel.tsx]: changed trigger type to boolean, left a note in clickup for later consideration.

**Review by stoodkev — APPROVED** [review]

**Pull request merged**